tests/Functional/Execution/AbstractTest.php

```diff
@@ -14,31 +13,161 @@
 use function Digia\GraphQL\Type\GraphQLObjectType;
 use function Digia\GraphQL\Type\GraphQLSchema;
 use function Digia\GraphQL\Type\GraphQLString;
+use function Digia\GraphQL\Type\GraphQLUnionType;
 
 class AbstractTest extends TestCase
 {
     // EXECUTE: HANDLES EXECUTION OF ABSTRACT TYPES
 
     /**
      * isTypeOf used to resolve runtime type for Interface
+     *
      * @throws \Digia\GraphQL\Error\InvariantException
      * @throws \Digia\GraphQL\Error\SyntaxErrorException
      */
-    public function testResolveFunctionForInterface()
+    public function testIsTypeOfUsedToResolveFunctionForInterface()
     {
+        $PetInterfaceType = GraphQLInterfaceType([
+            'name'   => 'Pet',
+            'fields' => [
+                'name' => ['type' => GraphQLString()]
+            ]
+        ]);
+
+        $DogType = GraphQLObjectType([
+            'name'       => 'Dog',
+            'interfaces' => [$PetInterfaceType],
+            'fields'     => [
+                'name'  => ['type' => GraphQLString()],
+                'woofs' => ['type' => GraphQLBoolean()],
+            ],
+            'isTypeOf'   => function ($obj) {
+                return $obj instanceof Dog;
+            }
+        ]);
+
+        $CatType = GraphQLObjectType([
+            'name'       => 'Cat',
+            'interfaces' => [$PetInterfaceType],
+            'fields'     => [
+                'name'  => ['type' => GraphQLString()],
+                'meows' => ['type' => GraphQLBoolean()],
+            ],
+            'isTypeOf'   => function ($obj) {
+                return $obj instanceof Cat;
+            }
+        ]);
+
         $schema = GraphQLSchema([
             'query' => GraphQLObjectType([
                 'name'   => 'Query',
                 'fields' => [
                     'pets' => [
-                        'type'    => GraphQLList(PetType()),
+                        'type'    => GraphQLList($PetInterfaceType),
                         'resolve' => function ($source, $args, $context, $info) {
-                            return [new Dog('Odie', true), new Cat('Garfield', false)];
+                            return [
+                                new Dog('Odie', true),
```

Comment:
Remember to add this file to the `autoload-dev` `files`.

Reply:
why?